### PR TITLE
Rtcp Unit Tests 

### PR DESCRIPTION
**Pull request by Varunaditya1**
### Description of changes:

This PR is for adding additional unit tests to improve coverage, by thoroughly testing various RTCP API's .
Also, The Unit-Tests are written from start and keeping the format same as the previous written unit-tests of other files. 

The following test cases have been added:
1. `test_rtcpSerializeReceiverReport`
2. `test_rtcpDeSerializePacket_WrongVersion`
3. `test_rtcpDeSerializePacket_MalformedPacked`
4. `test_rtcpDeSerializePacket`
5. `test_rtcpParsePliPacket`
6. `test_rtcpParseSliPacket`
7. `test_rtcpParseRembPacket_InvalidPacket`
8. `test_rtcpParseRembPacket_MalformedPacket`
9. `test_rtcpParseRembPacket_InvalidSSRCLength`
10. `test_rtcpParseRembPacket`
11. `test_rtcpParseSenderReport_OutOfMemory`
12. `test_rtcpParseSenderReport_MalformedPacket`
13. `test_rtcpParseSenderReport`
14. `test_rtcpParseReceiverReport_OutOfMemory`
15. `test_rtcpParseReceiverReport_MalformedPacket`
16. `test_rtcpParseReceiverReport`
17. `test_rtcpParseNackPacket`
18. `test_rtcpParseNackPacket_EmptySequenceList`
19. `test_rtcpParseNackPacket_OutOfMemory`

### Test Steps

```
git submodule update --init --recursive --checkout test/CMock
cmake -S test/unit-test -B build/ -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Debug -DBUILD_CLONE_SUBMODULES=ON -DCMAKE_C_FLAGS='--coverage -Wall -Wextra -Werror -DNDEBUG -DLIBRARY_LOG_LEVEL=LOG_DEBUG'
make -C build all
cd build
ctest -E system --output-on-failure
make coverage
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test/rtcp_api/rtcp_api_utest.c [hunk 1]
```diff
@@ -430,6 +498,133 @@ void test_rtcpDeSerializePacket_SmallPacket( void )
...
+                                     &( rtcpPacket ) );
+
+    TEST_ASSERT_EQUAL( RTCP_RESULT_WRONG_VERSION,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate RTCP DeSerialize fail Packet functionality for Malformed Packet.
```
Comment (ActoryOu) [inline test/unit-test/rtcp_api/rtcp_api_utest.c:536]: Which part of the packet content is malformed in this test case?

File: test/unit-test/rtcp_api/rtcp_api_utest.c [hunk 2]
```diff
@@ -719,66 +1025,412 @@ void test_rtcpParseRembPacket_BadParams( void )
...
+
+    rtcpPacket.payloadLength = 20;
+    rtcpPacket.pPayload = rembPacketPayload;
+    rtcpPacket.header.packetType = RTCP_PACKET_PAYLOAD_FEEDBACK_REMB;
+
+    result = Rtcp_ParseRembPacket( &( context ),
+                                   &( rtcpPacket ),
+                                   &( rtcpRembPacket ) );
+
+    TEST_ASSERT_EQUAL( RTCP_RESULT_OK,
```
Comment (ActoryOu) [inline test/unit-test/rtcp_api/rtcp_api_utest.c:1214]: Could you add a check on the result in rtcpRembPacket in pass case?

File: test/unit-test/rtcp_api/rtcp_api_utest.c (hunk 1 above)
Comment (Varunaditya1) [reply test/unit-test/rtcp_api/rtcp_api_utest.c:536]: The serialized Packet Length is smaller than expected . Hence the packet is malformed, input content is fine.

Comment (ActoryOu) [reply test/unit-test/rtcp_api/rtcp_api_utest.c:536]: Could you help add a comment for it?

File: test/unit-test/rtcp_api/rtcp_api_utest.c (hunk 2 above)
Comment (Varunaditya1) [reply test/unit-test/rtcp_api/rtcp_api_utest.c:1214]: Done, in the later commit.

**Review by ActoryOu — DISMISSED** [review]

**Review by aggarg — APPROVED** [review]

**Review by kar-rahul-aws — APPROVED** [review]

**Pull request merged**